libraries/AP_AHRS/AP_AHRS_DCM.cpp

```diff
@@ -1160,3 +1167,50 @@ bool AP_AHRS_DCM::pre_arm_check(bool requires_position, char *failure_msg, uint8
     }
     return true;
 }
+
+/*
+  relative-origin functions for fallback in AP_InertialNav
+*/
+bool AP_AHRS_DCM::get_origin_fallback(Location &ret) const
+{
+    ret = last_origin;
+    if (ret.is_zero()) {
+        // use home if we never have had an origin
+        ret = AP::ahrs().get_home();
+    }
```

Comment:
What are the implications for having restored this from the watchdog persistent data?

Should we assign home into `_last_origin` in the case we're forced to return home?  It's not expected to change, after all.

This could also check that `_home_is_set` and potentially that `_home_is_locked`.

Reply:
why check those? Just checking for is_zero() really should be enough